### ISSUE TITLE
Implement a TF2 version of `GPT2ForSequenceClassification`

### ISSUE THREAD

**Issue by LysandreJik** [huggingface/transformers#7622]
# 🚀 Feature request

The architecture `GPT2ForSequenceClassification` was added in #7501 in PyTorch. It would be great to have it in TensorFlow as well.

Below is a list of items to follow to make sure the integration is complete:

- Implement `TFGPT2ForSequenceClassification` in `modeling_tf_gpt2.py`
- Test that architecture in `tests/test_modeling_tf_gpt2.py`
- Add that architecture to `__init__.py` and ` docs/source/model_doc/gpt2.rst`.

Taking a look at the code changes in #7501 would be a good start, as this PR would essentially be a TF2 copy of it.

A very good first issue to get acquainted with the library and its architectures!

**Comment by y2s82** [comment]
May I take a crack at this? :)

**Comment by LysandreJik** [comment]
Yes, you may! Thanks @y2s82!

**Comment by spatil6** [comment]
HI @LysandreJik , I have partialy completed implementation for above feature request. I have observed some different/unexpected behavior in Tensorflow implementation of this model, while model initilization below default argument, Please see below. (I have put the print sentment in 'call' method)


![image](https://user-images.githubusercontent.com/6419011/97809402-adb42780-1c92-11eb-8883-7ba244f195e2.png)

I have cross checked above behavior in another TF implementation, it is same.

As default argument having batch size of 3, and there is no padding token, model initialization gets failed for assertion check where batch size >1 then padding token get checked.

    assert (
            self.config.pad_token_id is not None or batch_size == 1
        ), "Cannot handle batch sizes > 1 if no padding token is defined."


I'm thinking of putting "Label not None" check, if it is true, then only will calculate loss and return loss and  output. 

let me know, what should be done here, or is there more appropriate way to handle this default arguments at model initiliazation.

**Comment by LysandreJik** [comment]
Hi, I'm not sure I completely understand what's the issue here. Could you provide more details, maybe in a new issue so that we can identify if it's a bug or the expected behavior?

**Comment by LysandreJik** [comment]
Feel free to open a PR
